src/topology/compact_open.lean

```diff
@@ -46,7 +46,23 @@ variables [topological_space α] [topological_space β] [topological_space γ]
 
 def compact_open.gen (s : set α) (u : set β) : set C(α,β) := {f | f '' s ⊆ u}
 
--- The compact-open topology on the space of continuous maps α → β.
+variables (β)
+/-- For a fixed `s : set α`, the set of subsets of `C(α, β)` of the form `{f | f '' s ⊆ u}`, as `u`
+varies over the open subsets of `β`. -/
+def uniform_on.gen (s : set α) : set (set C(α, β)) :=
```

Comment:
I can't think of a good name here but I agree with your remarks on Zulip that we probably shouldn't use the work `uniform` (despite the eventual connection with uniform convergence when `β` is uniform).

Maybe simply `compact_open.exist_gen`?

Reply:
I would rather use the name `[topology_name]`.gen, where `topology_name` is the name we settle on for the topology I've currently called `uniform_on` ... let me know if you have any ideas for that!

Reply:
> I would rather use the name `[topology_name]`.gen, where `topology_name` is the name we settle on 

Oh I like this suggestion! Still hard to come up with a good name. _Maybe_ `pseudo_uniform_on.gen`?

Reply:
I am actually having doubts about whether the current `uniform_on` is actually the topology of uniform convergence -- it may be that actually the right definition is something like
```lean
def uniform_on.gen (s : set α) : set (set C(α, β)) :=
{m | ∃ (s' : set α) (hs' : is_compact s) (hss' : s' ⊂ s) (u : set β) (hu : is_open u),
  m = compact_open.gen s u}
```
(but the Inf property should still hold for this).  You are planning to try to write down the uniform structure and check its compatibility, right?  If so we'll find out for sure :)

If this turns out to be correct, maybe the right name for the topology is just something like `compact_open_on`, so the main useful results are (1)
```
compact_open = ⨅ (s : set α) (hs : is_compact s), compact_open_on β s :=
```
and (2) for `β` a uniform space and `s` compact, `compact_open_on β s` can be canonically uniformized.